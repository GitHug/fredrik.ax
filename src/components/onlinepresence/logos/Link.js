import React, { Component } from 'react';

class Link extends Component {
  render() {
    return (
      <a href={this.props.url} target={'_blank'}>
        {this.props.content}
      </a>
    );
  }
}

export default Link;
