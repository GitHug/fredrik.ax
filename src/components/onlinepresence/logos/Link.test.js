import React from 'react';
import ReactDOM from 'react-dom';
import Link from './Link';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Link url={'url'} content={<div>test</div>} />, div);
});

it('renders an anchor tag containing an url and an image', () => {
  const component = renderer.create(<Link url={'url'} content={'content'} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree.props.href).toBe('url');
  expect(tree.children[0]).toBe('content');
});
