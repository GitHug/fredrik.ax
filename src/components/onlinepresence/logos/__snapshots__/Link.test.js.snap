// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an anchor tag containing an url and an image 1`] = `
<a
  href="url"
  target="_blank"
>
  content
</a>
`;
